### ISSUE TITLE
Parkside PMRDA 20-Li A1 - None is not a valid LawnMowerActivity

### ISSUE THREAD

**Issue by nima-1102** [make-all/tuya-local#2139]
**Describe the bug**
The status of the Parkside Mähroboter PMRDA 20-Li A1 intermittently becomes unknown. This affects the lawn_mower, Rain mode, Running time, and Cover Entity statuses. The issue resolves temporarily upon reloading the device via the integrations GUI.

**To Reproduce**
1. Integrate the Parkside Mähroboter PMRDA 20-Li A1 with Home Assistant using the Tuya Local component.
2. Observe the status of the lawn_mower, Rain mode, Running time, and Cover Entity.
3. Notice that the statuses become unknown randomly at different times.
4. Reload the device via the integrations GUI to temporarily resolve the issue.

**Expected behavior**
The entities should consistently show the correct status without requiring manual reload.

**Additional context**
Logs when the problem occurs:
```
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/tuya_local/lawn_mower.py:59
integration: Tuya Local (documentation, issues)
First occurred: 07:50:44 (385 occurrences)
Last logged: 11:13:15

Error doing job: Exception in callback Entity._async_write_ha_state_from_call_soon_threadsafe() (None)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 998, in _async_write_ha_state_from_call_soon_threadsafe
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1133, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1068, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1013, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/lawn_mower/__init__.py", line 89, in state
    if (activity := self.activity) is None:
                    ^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/lawn_mower.py", line 59, in activity
    return LawnMowerActivity(self._activity_dp.get_value(self._device))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/enum.py", line 757, in __call__
    return cls.__new__(cls, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/enum.py", line 1171, in __new__
    raise ve_exc
ValueError: None is not a valid LawnMowerActivity
```
```
This error originated from a custom integration.

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:560
integration: Tuya Local (documentation, issues)
First occurred: 07:45:37 (13 occurrences)
Last logged: 07:50:19

Failed to fetch device status for Parkside Mähroboter
```

Log when reloading the device:
```
This error originated from a custom integration.

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:230
integration: Tuya Local (documentation, issues)
First occurred: 11:13:20 (1 occurrences)
Last logged: 11:13:20

Parkside Mähroboter receive loop has terminated
```

Version Info:
```
Tuya Local: 2024.7.0
Core: 2024.7.3
Supervisor: 2024.06.2
Operating System: 12.4
Frontend: 20240710.0
```

**Comment by make-all** [comment]
There doesn't appear to be a bug here. Your device goes offline.  The receive loop terminating is expected when you reload the integration.

If you want to improve the recovery time of your device when it goes offline, then ensure the protocol version and IP address are both set.

**Comment by nima-1102** [comment]
Thank you very much for creating the integration.

The IP address was fixed in the device config, I have now changed the protocol from auto to 3.3.

The device was not accessible for 6 minutes, I have a ping sensor for the device, so after that the device was accessible again by HomeAssitant for ~3.5 hours until I reloaded the integration.

Looks like a bug to me, since the integration should normally automatically recognize the device as accessible again after this period of time.
